build.gradle

```diff
@@ -57,18 +57,16 @@ wrapper {
 }
 
 ext {
-    guavaVersion = "11.0.2"
     kafkaVersion = "2.2.0"
     slf4jVersion = "2.0.12"
     log4jVersion = "2.23.0"
-    elasticSearchVersion = "7.4.0"
-    elasticClientVersion = "7.17.0"
+    elasticJavaClientVersion = "7.17.18"
```

Comment:
If I understand [this](https://github.com/elastic/elasticsearch-java/?tab=readme-ov-file#compatibility) correctly 
> client supports communicating with greater or equal minor versions of Elasticsearch without breaking

this PR makes connector plugin compatible with ES 7.x and requires `elasticJavaClientVersion = "8.x.y"` to make it compatible to ES 8.x?

Reply:
No, the `8.x` version of Java client is not required for `8.x` compatibility. The `7.17.18` is enough.
There is two test configurations running, the default which starts `8.x` in container and another defined in the Gradle build file that runs same tests with `7.x` container.